cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -273,6 +304,21 @@ private void removeTrigger(Table table, TriggerKey key) {
     table.delete(TRIGGER_KEY, col);
   }
 
+  private JobDetail readJob(JobKey jobKey) {
```

Comment:
Yes, while trying to upgrade, we need to check if the upgraded job key already exists. If it exists, we don't replace it.